code/modules/persistence/storage/smartfridge.dm

```diff
@@ -79,6 +70,16 @@
 	name = "fruit storage lossy"
 	go_missing_chance = 12.5 // 10% loss between rounds
 
+/datum/persistent/storage/smartfridge/produce/generate_items(var/list/L)			// Mostly same as storage/generate_items() but without converting string to path
```

Comment:
I haven't looked at the rest. Cute as this is-

- Persistent storage is a terrible mistake.
- The base implementation does text=type checking for a reason.
- The base implementation does not check *count* on write or read, but *really* should.
- ***Persistent storage is a terrible mistake***.